### ISSUE TITLE
Cannor load DLL on WIndows

### ISSUE THREAD

**Issue by giohappy** [ideoforms/signalflow#124]
Python 3.11.9 on Windows 11.
Signalflow fails when imprted.

This is the output when traced with windlltracer

**Comment by ideoforms** [comment]
I had another report of this which was eventually resolved… Do you happen to be running a 32-bit version of Python or otherwise in a 32-bit environment? The library is currently only built for 64-bit architectures, so I suspect this might be the issue.

**Comment by giohappy** [comment]
@ideoforms I'm on a 64 bit machine with Python 64 bit. I also hoped that was the reason, but no.

**Comment by hockinsk** [comment]
This issue came back for me in a new venv. Looking at signalflow.pyd in Dependencies, signalflow's  __init__.py is failing because the paths to the the .dlls don't seem to have resolved perhaps? I would have thought the .pyd should be resolving the paths to: \Lib\site-packages\signalflow.libs\name.dll

One quick resolution is simply copy the .dll's from signalflow.libs and place them in the \Lib\site-packages\signalflow directory, then: from signalflow import * should work.

![image](https://github.com/user-attachments/assets/ecff2034-2630-45b5-a906-288cc4deefdd)

**Comment by hockinsk** [comment]
Actually, it makes no difference copying then to signalflow dir, it still doesn't find the three .dll.

**Comment by hockinsk** [comment]
I thought I'd have a nose in signalflow.pyd hex to see what the dll paths were and found a lot of ascii error output related to JACK? Didn't expect to see that, i'm sure it's not related to this dll issue, but wasn't what I expected to find so sharing anyway.

I just removed all C++ redistributable, removed python, removed visual studio, installed python, installed signalflow but still errors the module can't be found.

![image](https://github.com/user-attachments/assets/5e6a2de5-4ce0-416f-acb1-643b8aa8b3c7)

**Comment by hockinsk** [comment]
Rolling back to signalflow 0.4.4 works on same system. I notice there's a soundio .dll in soundflow.libs that doesn't exist in the 0.5.1 latest version, but assume this is handled differently now?

**Comment by ideoforms** [comment]
That's so strange. You're right, soundio.dll shouldn't be needed anymore. I'm travelling at the moment, but will take a look at the DLLs sometime when I'm back at my Windows computer next week.

**Comment by hockinsk** [comment]
Great stuff. Probably looking at differences between 0.4.4 > 0.5.1 pip upgrade versus a fresh 0.5.1 install might show the issue because the upgrade path (using .whl instead of pip) seems to work.

**Comment by balintlaczko** [comment]
I bumped into the same issue now using a conda env that used python 3.10. I tried to perform @hockinsk's hack, but then noticed that v0.4.4 is cpython 12 only. Then I scratched the whole env and recreated it with python 3.12, and signalflow 0.5.1 (latest at the time of writing) and everything works. So, afaik it is not the soundio.dll, because on my system no version other than 0.5.1 was ever installed. There seems to be something in the cp310_amd64 wheel that has an issue.

**Comment by ideoforms** [comment]
@balintlaczko So this was Windows/conda/py3.10? Do you still have that env? I'd be interested in some diags if you're still able to reproduce the problem, particularly `pip show signalflow` and the output of `pip debug -v` (which may be long but that's ok!).

**Comment by hockinsk** [comment]
I I were to guess, it's JACK Audio Connection kit erroring while compiling signalflow.pyd and it's corrupted something, this is why upgrading from a working 0.4.4 to 0.5.1 seems to be fine, but installing directly to  0.5.1 simply isn't??

**Comment by balintlaczko** [comment]
@hockinsk I cannot install v0.4.4 on python 3.10, and it[ seems like the release is only for Cpython 3.12](https://pypi.org/project/signalflow/0.4.4/#files)? But if I use python 3.12 then the latest version v.5.0.1 also works without issues.

**Comment by ideoforms** [comment]
Yeah, v0.4.4 was a Windows test release that only targeted 3.12. v0.5.1 is the first proper cross-Python release for Windows.

@balintlaczko Do you think you would be able to try running [this utility](https://github.com/lucasg/Dependencies) on the different signalflow .pyd files in your 3.10 vs 3.12 conda envs? (A .pyd file is really just a DLL in disguise on Windows). This might shed light on which DLL it's failing to find. 

@hockinsk The warning messages about JACK when you look inside the binary are simply because SignalFlow now compiles with [miniaudio](https://miniaud.io/) as an include, so these are benign and expected.

**Comment by balintlaczko** [comment]
@ideoforms sure thing! This is what it shows for the failing env ("sftest") with python 3.10

![Image](https://github.com/user-attachments/assets/983b657c-36d2-43ae-a522-77275d9d9100)

And this is what I get in the env where it works ("pixasonics") with python 3.12

![Image](https://github.com/user-attachments/assets/6998be23-d262-4151-b029-3c4696b5e75d)

Not sure hoow to interpret this, but hope it helps!

**Comment by ideoforms** [comment]
Interesting! The 3.10 build seems to be looking for `python39.dll`, which is definitely not right. I'll investigate whether this is a problem with things not properly getting configured or cleaned up on the build machine. 

Thanks so much for your help @balintlaczko, I will try to get a new build out as soon as I have the opportunity.

**Comment by balintlaczko** [comment]
Thanks a lot!!! (But as for my case, no rush! I'm fine using 3.12 in the env.)

**Comment by ideoforms** [comment]
Thanks for all your help with this. I tracked this down to a silly issue in the build process, in which it sometimes bundled the wrong cached version of the .pyd file. 🤪

This should now be fixed!

Could you please try installing the latest (0.5.2, Windows-only) release and let me know if this resolves the issue @balintlaczko @hockinsk @giohappy?

**Comment by balintlaczko** [comment]
Thanks for the fix, @ideoforms, this solved it for me! In the same "sftest" env that was failing before under python 3.10 I uninstalled the previous signalflow 0.5.1 and installed the new 0.5.2, and then imported signalflow without the dll error. Nice!
